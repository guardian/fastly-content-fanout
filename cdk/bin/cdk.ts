import 'source-map-support/register';
import { GuRoot } from '@guardian/cdk/lib/constructs/root';
import { EventbridgeToFanout } from '../lib/eventbridge-to-fanout';

const FRONTS_STACK = 'cms-fronts';
const CAPI_STACK = 'content-api-crier-v2'; // crier also has a preview stack (content-api-crier-v2-preview) but we don't think this is needed right now
const app = new GuRoot();

const env = { region: 'eu-west-1' };

// fronts
new EventbridgeToFanout(app, 'EventBridgeToFanout-eu-west-1-fronts-CODE', {
	stack: FRONTS_STACK,
	stage: 'CODE',
	env,
});
new EventbridgeToFanout(app, 'EventBridgeToFanout-eu-west-1-fronts-PROD', {
	stack: FRONTS_STACK,
	stage: 'PROD',
	env,
});

// capi
new EventbridgeToFanout(app, 'EventBridgeToFanout-eu-west-1-capi-CODE', {
	stack: CAPI_STACK,
	stage: 'CODE',
	env,
	// withKinesisStreamArnAsPipeSource: 'foo',
});

new EventbridgeToFanout(app, 'EventBridgeToFanout-eu-west-1-capi-PROD', {
	stack: CAPI_STACK,
	stage: 'PROD',
	env,
	withKinesisStreamArnAsPipeSource: 'bar',
});
