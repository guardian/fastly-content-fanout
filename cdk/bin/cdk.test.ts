import {riffRaff} from "./cdk";

describe('The riff-raff output YAML', () => {
  it('matches the snapshot', () => {
    // @ts-ignore
    const outdir = riffRaff.outdir; // this changes for every test execution and best not to change cdk.ts too much
    const riffRaffYaml = riffRaff.toYAML().replaceAll(outdir, 'cdk.out');
    expect(riffRaffYaml).toMatchSnapshot();
  });
});
