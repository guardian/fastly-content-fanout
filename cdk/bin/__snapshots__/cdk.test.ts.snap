// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The riff-raff output YAML matches the snapshot 1`] = `
"allowedStages:
  - CODE
  - PROD
deployments:
  cfn-eu-west-1-cms-fronts-eventbridge-to-fanout:
    type: cloud-formation
    regions:
      - eu-west-1
    stacks:
      - cms-fronts
    app: eventbridge-to-fanout
    contentDirectory: cdk.out
    parameters:
      templateStagePaths:
        CODE: EventBridgeToFanout-eu-west-1-fronts-CODE.template.json
        PROD: EventBridgeToFanout-eu-west-1-fronts-PROD.template.json
  cfn-eu-west-1-content-api-fastly-cache-purger-eventbridge-to-fanout:
    type: cloud-formation
    regions:
      - eu-west-1
    stacks:
      - content-api-fastly-cache-purger
    app: eventbridge-to-fanout
    contentDirectory: cdk.out
    parameters:
      templateStagePaths:
        CODE: EventBridgeToFanout-eu-west-1-capi-CODE.template.json
        PROD: EventBridgeToFanout-eu-west-1-capi-PROD.template.json
  fastly-C@E-package:
    type: fastly-compute
    app: fastly-content-fanout
    contentDirectory: fastly-C@E-package
    parameters: {}
    regions:
      - eu-west-1
    stacks:
      - mobile
"
`;
